base/nvti.c

```diff
@@ -159,6 +159,141 @@ vtref_text (const vtref_t *r)
   return r ? r->ref_text : NULL;
 }
 
+/* VT severities */
+
+/**
+ * @brief The structure for a severity of a VT.
+ *
+ * VTs can have one or several severities.
+ */
+typedef struct vtseverity
+{
+  gchar *type;   ///< Severity type ("cvss_base_v2", ...)
+  gchar *origin; ///< Optional: Where does the severity come from
+                 ///< ("CVE-2018-1234", "Greenbone Research")
+  int date; ///< Timestamp in seconds since epoch, defaults to VT creation date.
+  int score;    ///< The score derived from the value in range [0-100]
+  gchar *value; ///< The value which corresponds to the type.
+} vtseverity_t;
+
+/**
+ * @brief Create a new vtseverity structure filled with the given values.
+ *
+ * @param type The severity type to be set.
+ *
+ * @param origin The origin reference to be set, can be NULL.
+ *
+ * @param value The value corresponding to the type.
+ *
+ * @return NULL in case the memory could not be allocated.
+ *         Else a vtref structure which needs to be
+ *         released using @ref vtref_free .
+ */
+vtseverity_t *
+vtseverity_new (const gchar *type, const gchar *origin, int date, int score,
+                const gchar *value)
+{
+  vtseverity_t *s = g_malloc0 (sizeof (vtseverity_t));
+
+  if (type)
+    s->type = g_strdup (type);
+  if (origin)
+    s->origin = g_strdup (origin);
+  s->date = date;
```

Comment:
i really think date should be optional too

Reply:
nevermind. it is an int and therefore can be -1 for not set or something like that.